lib/Cake/View/Helper/FormHelper.php

```diff
@@ -1292,7 +1292,7 @@ protected function _maxLength($options) {
 		if ($autoLength &&
 			in_array($options['type'], array('text', 'textarea', 'email', 'tel', 'url', 'search'))
 		) {
-			$options['maxlength'] = $fieldDef['length'];
+			$options['maxlength'] = array_sum(explode(',', $fieldDef['length']));
```

Comment:
`DECIMAL(10,5)` is still of length 10! It means that out of 10 numbers, 5 should be decimals.